contracts/TokenManager.sol

```diff
@@ -189,6 +273,262 @@ contract TokenManager is System, IApplication {
+    elements[1] = syncSynPackage.bep20Addr.encodeAddress();
+    elements[2] = uint256(syncSynPackage.bep2Symbol).encodeUint();
+    elements[3] = syncSynPackage.bep20Supply.encodeUint();
+    elements[4] = syncSynPackage.syncFee.encodeUint();
+    elements[5] = uint256(syncSynPackage.expireTime).encodeUint();
+    return elements.encodeList();
+  }
+
+  function decodeSyncSynPackage(bytes memory msgBytes) internal pure returns(SyncSynPackage memory, bool) {
+    SyncSynPackage memory syncSynPackage;
+    RLPDecode.Iterator memory iter = msgBytes.toRLPItem().iterator();
+    bool success = false;
+    uint256 idx=0;
+    while (iter.hasNext()) {
+      if (idx == 0)      syncSynPackage.syncSender  = iter.next().toAddress();
+      else if (idx == 1) syncSynPackage.bep20Addr   = iter.next().toAddress();
+      else if (idx == 2) syncSynPackage.bep2Symbol  = bytes32(iter.next().toUint());
+      else if (idx == 3) syncSynPackage.bep20Supply = iter.next().toUint();
+      else if (idx == 4) syncSynPackage.syncFee     = iter.next().toUint();
+      else if (idx == 5) {
+        syncSynPackage.expireTime = uint64(iter.next().toUint());
+        success = true;
+      }
+      else break;
+      idx++;
+    }
+    return (syncSynPackage, success);
+  }
+
+  function decodeSyncAckPackage(bytes memory msgBytes) internal pure returns(SyncAckPackage memory, bool) {
+    SyncAckPackage memory syncAckPackage;
+    RLPDecode.Iterator memory iter = msgBytes.toRLPItem().iterator();
+    bool success = false;
+    uint256 idx=0;
+    while (iter.hasNext()) {
+      if (idx == 0)      syncAckPackage.syncSender   = iter.next().toAddress();
+      else if (idx == 1) syncAckPackage.bep20Addr    = iter.next().toAddress();
+      else if (idx == 2) syncAckPackage.syncFee      = iter.next().toUint();
+      else if (idx == 3) {
+        syncAckPackage.errorCode  = uint8(iter.next().toUint());
+        success = true;
+      }
+      else break;
+      idx++;
+    }
+    return (syncAckPackage, success);
+  }
+
+  function sync(address bep20Addr, uint64 expireTime) payable public returns (bool) {
+    bytes32 bep2Symbol = ITokenHub(TOKEN_HUB_ADDR).getBep2SymbolByContractAddr(bep20Addr);
+    require(bep2Symbol != bytes32(0x00), "not bound");
+    require(boundByMirror[bep20Addr], "not bound by mirror");
+    uint256 miniRelayFee = ITokenHub(TOKEN_HUB_ADDR).getMiniRelayFee();
+    require(msg.value%TEN_DECIMALS == 0 && msg.value>=syncFee.add(miniRelayFee), "msg.value must be N * 1e10 and no less sum of miniRelayFee and syncFee");
+    require(expireTime>=block.timestamp + 120 && expireTime <= block.timestamp + 86400, "expireTime must be two minutes later and one day earlier");
+    uint256 totalSupply = IBEP20(bep20Addr).totalSupply();
+    uint8 decimals = IBEP20(bep20Addr).decimals();
+    require(convertToBep2Amount(totalSupply, decimals) <= MAX_BEP2_TOTAL_SUPPLY, "too large total supply");
+
+    address(uint160(TOKEN_HUB_ADDR)).transfer(msg.value.sub(syncFee));
```

Comment:
Shoud transfer relayer fee here, not `syncFee `.

Reply:
What relayer fee? The minimum relayer fee? The fee should not be refunded.

Reply:
I mean here did not transfer the `minimum relayer fee`.